src/main/drivers/display.c

```diff
@@ -85,6 +97,16 @@ int displayWrite(displayPort_t *instance, uint8_t x, uint8_t y, const char *s)
 
 int displayWriteWithAttr(displayPort_t *instance, uint8_t x, uint8_t y, const char *s, textAttributes_t attr)
 {
+    if (attr & ~instance->supportedTextAttributes) {
+        // We can't overwrite s, so we write char by char
```

Comment:
Can't we reserve a buffer large enough and process the string in chunks? We fill the buffer, once it's filled - we flush it and continue with the rest of `s`.